Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -1991,7 +1995,17 @@ extension URL {
 
 #if !NO_FILESYSTEM
     private static func isDirectory(_ path: String) -> Bool {
+#if os(Windows)
+        return (try? path.withNTPathRepresentation {
+            var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+            guard GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) else {
+                return false
+            }
+            return faAttributes.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY == FILE_ATTRIBUTE_DIRECTORY
+        }) ?? false
+#else
         return FileManager.default._fileStat(path)?.isDirectory ?? false
```

Comment:
I wonder if we should just use `FileManager.default.fileExists(atPath:isDirectory:)` which is already cross platform. Do you think that'd be ok here? @jrflat do you see any issues with that?

Reply:
Yeah, that does seem nicer.

Reply:
@jmschonfeld should I just revise this PR to do that?

Reply:
Using `fileExists(atPath:isDirectory:)` sounds good to me!